### DISCUSSION TITLE
C# dotnet - Error while authenticating. Extended authentication handler is not yet supported

### DISCUSSION THREAD

**Discussion by saud-saeed** [dotnet/MQTTnet#1786] [Q&A] [answered]
Hello,

I'm encountering an issue while attempting to connect to AWS IoT Core from my C# Web API in .NET 7.0. Interestingly, the same setup was functioning perfectly a few days ago. The error message I'm now receiving is as follows:

MQTTnet.Adapter.MqttConnectingFailedException: 'Error while authenticating. Extended authentication handler is not yet supported'

My code is as below:
var mqttClient = factory.CreateMqttClient();
                var options = new MqttClientOptionsBuilder()
                        .WithTcpServer(IotEndpoint, BrokerPort)
                        .WithClientId(ClientId)
                        .WithCleanSession()
                        //.WithCommunicationTimeout(TimeSpan.FromSeconds(60))
                        .WithTls(new MqttClientOptionsBuilderTlsParameters
                        {
                            UseTls = true,
                            SslProtocol = System.Security.Authentication.SslProtocols.Tls12,
                            IgnoreCertificateRevocationErrors = true,
                            IgnoreCertificateChainErrors = true,
                            AllowUntrustedCertificates = true,
                            Certificates = new List<X509Certificate>
                            {
                                pfxCert
                            }
                        })
                        .Build();

**Comment by SeppPenner** [comment]
I had the same issue right now and it was caused by two clients having the same client id. Can you check if this is the case?

**Reply by saud-saeed** [reply]
For my scenario, I am dealing with two clients that have distinct client IDs. If you've managed to resolve this issue, could you kindly provide some more details or share a snippet of the code you used? This would be immensely helpful.

**Reply by SeppPenner** [reply]
In my case, there were two clients with the same client id... After changing that, it worked.

**Reply by saud-saeed** [reply]
Could you please share a snippet of the code you used while connecting to AWS IoT MQTT.

**Comment by RashiqueProgbiz** [comment]
Have you got the solution?

**Reply by saud-saeed** [reply, marked as answer]
Yes, this issue has been resolved now. To resolve the IoT connectivity problem, I followed these steps:

1. Create a new certificate in the AWS IoT Core console.
2. Download the certificate and key files.
3. Replace the content of the private key and certificate PEM files in the Web API project.
4. After performing these steps, the connection was successfully established, and we are now able to send and receive messages from IoT once again.

**Reply by RashiqueProgbiz** [reply]
> Yes, this issue has been resolved now. To resolve the IoT connectivity problem, I followed these steps:
> 
> 1. Create a new certificate in the AWS IoT Core console.
> 2. Download the certificate and key files.
> 3. Replace the content of the private key and certificate PEM files in the Web API project.
> 4. After performing these steps, the connection was successfully established, and we are now able to send and receive messages from IoT once again.

Thank you so much for providing the solution.
